### PR TITLE
Add EasyDel Qwen3-14B Base and Qwen3-32B LoRA fine-tuning experiments (JAX, Galaxy)

### PR DESCRIPTION
**Pull request by ndimicTT**
### Issue

N/A (extends #581 to Galaxy, no associated issue)

### Training Workload Setting

- **Model:** Qwen3-14B (Qwen/Qwen3-14B-Base) and Qwen3-32B (Qwen/Qwen3-32B)
- **Strategy:** LoRA (rank 32, q_proj + v_proj), data + tensor parallel on attention/MLP/lm_head
- **Framework:** JAX / EasyDel
- **Task:** Instruction-style causal language modelling
- **Dataset:** SST-2 (GLUE) and Alpaca (tatsu-lab/alpaca)
- **Hardware:** WH Galaxy (32 chips)

### What's Changed

This PR extends the EasyDel/JAX LoRA experiment from #581 (QuietBox, Qwen3-4B) to Galaxy on larger Qwen3 models. It adds:

- A `galaxy/` config folder under `easydel/qwen/lora` with three configs, all on a `mesh [4, 8]/data,model` layout (DP × TP) across the 32-chip Galaxy:
  - `qwen3_14b_sst2.yaml` — Qwen3-14B, SST-2
  - `qwen3_14b_alpaca.yaml` — Qwen3-14B, Alpaca
  - `qwen3_32b_sst2.yaml` — Qwen3-32B, SST-2
- Removal of the per-step gradient statistics (global grad norm / max) computation and W&B logging from the fused train step, which required a device→host readback of the gradients each step.

#### Qwen 14B on sst2 dataset
qwen_14b_sst2_train
qwen_14b_sst2_val

#### Qwen 14B on Alpaca dataset
W B Chart 11_06_2026, 17_01_07
W B Chart 11_06_2026, 17_01_23

### Checklist

- [x] Main training workflow
- [ ] CPU testing
- [x] GPU testing
- [x] TT device testing
- [ ] CI testing
- [ ] Existing tests are passing
- [ ] Register test to CI
- [ ] Upload golden files if necessary

### CODE REVIEW

File: blacksmith/experiments/easydel/qwen/lora/galaxy/qwen3_14b_sst2.yaml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+mesh_shape: [4, 8]
+mesh_axis_names: ["data", "model"]
+input_sharding_dim: "data"
+apply_gqa_workaround: true
+
+# Tensor-parallel sharding patterns.
+# nn.LoRA wrapping moves the original q_proj/v_proj kernel from
+# self_attn.<proj>.kernel.value to self_attn.<proj>.base_module.kernel.value
+# and adds lora_a [in, rank] and lora_b [rank, out] siblings. The (?:\.base_module)?
+# group keeps the rule working whether or not LoRA is applied.
```
Comment (pglusacTT) [inline blacksmith/experiments/easydel/qwen/lora/galaxy/qwen3_14b_sst2.yaml]: Remove comment.

File: blacksmith/experiments/easydel/qwen/lora/train.py [hunk 2]
```diff
@@ -238,15 +239,15 @@ def _make_batch(input_ids, labels, attention_mask):
                     logger.log_metrics({"train/avg_window_loss": average_window_loss}, step=global_step, commit=False)
                     logger.info(
                         f"Epoch {epoch + 1}, Batch {batch_index + 1:3d}: "
-                        f"Loss = {current_loss:.4f} | Avg {config.steps_freq} = {average_window_loss:.4f} | "
-                        f"grad_norm = {gradient_norm:.4f}, grad_max = {gradient_max:.4f}"
+                        f"Loss = {current_loss:.4f} | Avg {config.steps_freq} = {average_window_loss:.4f}"
+                        # f"grad_norm = {gradient_norm:.4f}, grad_max = {gradient_max:.4f}"
```
Comment (pglusacTT) [inline blacksmith/experiments/easydel/qwen/lora/train.py]: Remove dead code.

File: blacksmith/tools/jax/easydel/train_steps.py [hunk 3]
```diff
@@ -107,13 +107,13 @@ def fused_train_step(
         updates, new_opt = tx.update(grads, opt_state, lora_params)
         new_params = optax.apply_updates(lora_params, updates)
         new_params = jax.tree.map(jax.lax.optimization_barrier, new_params)
-        leaves = jax.tree.leaves(grads)
-        grad_norm = jnp.sqrt(sum(jnp.sum(g**2) for g in leaves))
-        grad_max = jnp.max(jnp.stack([jnp.max(jnp.abs(g)) for g in leaves]))
+        # leaves = jax.tree.leaves(grads)
+        # grad_norm = jnp.sqrt(sum(jnp.sum(g**2) for g in leaves))
+        # grad_max = jnp.max(jnp.stack([jnp.max(jnp.abs(g)) for g in leaves]))
```
Comment (pglusacTT) [inline blacksmith/tools/jax/easydel/train_steps.py]: As well.

File: blacksmith/experiments/easydel/qwen/lora/galaxy/qwen3_14b_sst2.yaml (hunk 1 above)
Comment (ndimicTT) [reply blacksmith/experiments/easydel/qwen/lora/galaxy/qwen3_14b_sst2.yaml]: Removed.

File: blacksmith/tools/jax/easydel/train_steps.py (hunk 3 above)
Comment (ndimicTT) [reply blacksmith/tools/jax/easydel/train_steps.py]: Removed.

File: blacksmith/experiments/easydel/qwen/lora/train.py (hunk 2 above)
Comment (ndimicTT) [reply blacksmith/experiments/easydel/qwen/lora/train.py]: Removed.

**Review by pglusacTT — APPROVED** [review]
